feat: impact graphs

FelixMoelder opened the pull request:
Introduction of processing impact graphs using predictosaurus.

FelixMoelder committed:
feat: impact graphs

FelixMoelder committed:
fix: multiple minor fixes

FelixMoelder committed:
remove print cmd

FelixMoelder committed:
Merge branch 'master' into feat/impact_graphs

fxwiegand commented on workflow/envs/predictosaurus.yaml:
```diff
@@ -0,0 +1,5 @@
+  - conda-forge
+  - bioconda
+dependencies:
+  - predictosaurus =0.2.10
```
```suggestion
  - predictosaurus =0.4.0
```

fxwiegand commented on workflow/rules/impact_graphs.smk:
```diff
@@ -0,0 +1,88 @@
+        gff="resources/annotation.gff3",
+        graph="results/impact_graphs/{group}.freebayes.graphs.duckdb",
+    output:
+        "results/impact_graphs/{group}.paths.duckdb",
```
```suggestion
        "results/impact_graphs/{group}.scores.duckdb",
```

fxwiegand commented on workflow/rules/impact_graphs.smk:
```diff
@@ -0,0 +1,88 @@
+        "../envs/predictosaurus.yaml"
+    shell:
+        """
+        predictosaurus plot --input {input} --format html --output {output} 2> {log}
```
```suggestion
        predictosaurus plot --input {input} --output {output} 2> {log}
```

fxwiegand reviewed (commented):
I have left a few suggestions how to update stuff for the new score calculation in `v0.4.0`. I think otherwise we are good to give it a test spin!

FelixMoelder committed:
Apply suggestions from code review

Co-authored-by: Felix Wiegand <fxwiegand@wgdnet.de>

FelixMoelder committed:
Merge branch 'fix/minor_fixes' into feat/impact_graphs

FelixMoelder committed:
add to report

FelixMoelder committed:
Merge branch 'master' into feat/impact_graphs

FelixMoelder committed:
add impact graph spec

FelixMoelder committed:
add js file

fxwiegand committed:
Update predictosaurus version to 0.7.0

fxwiegand committed:
fix: Use input.obs to build predictosaurus aux entries (fixes paths for
s3 storage use-case)

fxwiegand committed:
Add min-vaf 0.10 to predictosaurus build

fxwiegand committed:
Update predictosaurus version to v0.7.1

fxwiegand committed:
Bump predictosaurus to 0.7.3

FelixMoelder committed:
fix datavzrd input

fxwiegand committed:
WIP: Handle JSON impact scores and update graphs

Convert impact score handling to structured JSON across the pipeline
(split-call-tables, data parser, Vega-Lite spec). Lower predictosaurus
thresholds, increase processing threads, and raise max-in-memory-rows.

fxwiegand committed:
WIP: Serialize impact_scores as list of sample objects

Produce impact_scores as an array of {score, sample, likelihood}
entries. Remove the Vega transforms that flattened/folded/calc'ed
impact_scores. Add parse_impact_scores to replace single quotes and
strip trailing commas before calling JSON.parse.

fxwiegand committed:
WIP: Prevent variable shadowing in format_scores

Rename list variable to score_list to avoid shadowing the loop variable

fxwiegand committed:
WIP: Fix impact score parsing and CSV output

Remove ad-hoc fixes in parse_impact_scores; expect valid JSON

Add csv import and write TSV without quoting (escape backslashes).
Aggregate impact_scores by concatenating lists (sum) and serialize
the grouped column with json.dumps.

fxwiegand committed:
Update Vega-Lite spec for impact graph

Set width and height and rename data source to "scores"
Change mark from tick to circle and set size to 100
Add x/y axis titles (Score, Likelihood) and add tooltip for score and likelihood

fxwiegand committed:
Handle empty impact scores and swap axes

Return [] when parse_impact_scores receives a falsy value to avoid
JSON.parse errors. Swap x/y encoding in the spec so score maps to y and
likelihood maps to x.

fxwiegand committed:
Update predictosaurus version

fxwiegand committed:
Update predictosaurus dependency to version 0.8.1

fxwiegand committed:
fix: Show haplotype scores with frequency ticks and horizontal bar plot

fxwiegand committed:
Update predictosaurus.yaml

fxwiegand committed:
Improve impact score vega lite plot

fxwiegand committed:
feat: Include haplotype with contained variants in impact score
visualization using predictosaurus 0.9

fxwiegand committed:
Adjust haplotype plot

fxwiegand committed:
feat: Convert impact graph to hconcat panels

Separate score bar and variant matrix into two panels.
Aggregate max score and frequency per haplotype and flatten
variants into rows, add transforms, tooltips, and shared y-scale.
Adjust marks, axes, encodings, and color defaults

fxwiegand committed:
Update predictosaurus to 0.9.1 for proper hgvsc notation in haplotypes

fxwiegand committed:
Adjust vega specs to new hgvsc form

fxwiegand committed:
Sort variants by position

fxwiegand committed:
Update predictosaurus version

fxwiegand committed:
Update predictosaurus.yaml

fxwiegand committed:
Bump predictosaurus to 0.10.0

FelixMoelder committed:
Merge branch 'master' into feat/impact_graphs

fxwiegand committed:
feat: Use per-sample frequency and supporting_reads for predictosaurus
v0.10.0

Detect sample names from columns ending with :frequency and strip them.
Read frequency and supporting_reads from '<sample>:frequency' and
'<sample>:supporting_reads' columns when building score dicts.

fxwiegand committed:
Merge branch 'feat/impact_graphs' of github.com:snakemake-workflows/dna-seq-varlociraptor into feat/impact_graphs

fxwiegand committed:
Display supporting reads and update predictosaurus version
